pandas/io/formats/excel.py

```diff
@@ -647,6 +650,12 @@ def write(self, writer, sheet_name='Sheet1', startrow=0,
         """
         from pandas.io.excel import ExcelWriter
         from pandas.io.common import _stringify_path
+        
+        num_rows, num_cols = df.shape
```

Comment:
`df` isn't defined here which is causing most if not all of your CI failures. I think you want `self.df`

Reply:
Fixed! .. new bug. 

New bug:
  AssertionError: Caused unexpected warning(s): [('PendingDeprecationWarning', 
  PendingDeprecationWarning("This method will be removed in future versions

Pushed my changes.

Reply:
I still think there's some kind of issue with my environment. 

I tried running the code again on my rollback branch. The code crashes with the same AssertionError above.

Can you post your environment.yml? I can try using that to execute the code.